diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/TypeAliasRule.kt

```diff
@@ -24,15 +26,15 @@ class TypeAliasRule(private val configRules: List<RulesConfig>) : Rule("type-ali
         emitWarn = emit
         isFixMode = autoCorrect
 
-        val config = TypeAliasConfiguration(configRules.getRuleConfig(TYPE_ALIAS)?.configuration ?: mapOf())
-        if (node.elementType == PROPERTY)
-            checkProperty(node, config)
+        if (node.elementType == TYPE_REFERENCE) {
+            checkProperty(node, TypeAliasConfiguration(configRules.getRuleConfig(TYPE_ALIAS)?.configuration ?: mapOf()))
+        }
     }
 
     private fun checkProperty(node: ASTNode, config: TypeAliasConfiguration) {
-        val typeReference = node.findChildByType(TYPE_REFERENCE) ?: return
-        if (typeReference.textLength > config.typeReferenceLength)
-            TYPE_ALIAS.warn(configRules, emitWarn, isFixMode, "too long type reference", typeReference.startOffset)
+        if (node.textLength > config.typeReferenceLength)
+            if (node.findAllNodesWithSpecificType(LT).size > 1 || node.findAllNodesWithSpecificType(VALUE_PARAMETER).size > 1)
```

Comment:
What does `VALUE_PARAMETER` correspond to in this case? Why is count of `LT` not enough?

Reply:
I added KDoc for this case.